content/blog/2018/egress-mongo/index.md

```diff
@@ -0,0 +1,1120 @@
+      hosts:
+      - $MONGODB_HOST
+      gateways:
+      - mesh
+      - istio-egressgateway
+      tls:
+      - match:
+        - gateways:
+          - mesh
+          port: $MONGODB_PORT
+          sni_hosts:
+          - $MONGODB_HOST
+        route:
+        - destination:
+            host: istio-egressgateway.istio-system.svc.cluster.local
+            subset: mongo
+            port:
+              number: 443
+      - match:
+        - gateways:
+          - istio-egressgateway
+          port: 443
+          sni_hosts:
+          - $MONGODB_HOST
+        route:
+        - destination:
+            host: $MONGODB_HOST
+            port:
+              number: $MONGODB_PORT
+          weight: 100
+    EOF
+    {{< /text >}}
+
+1.  Refresh the web page of the application again and verify that the ratings are still displayed correctly.
+
+1.  Check the statistics of the egress gateway's Envoy and see a counter that corresponds to your
+    requests to the MongoDB service. If Istio is deployed in the `istio-system` namespace, the command to print the
+    counter is:
+
+    {{< text bash >}}
+    $ kubectl exec -it $(kubectl get pod -l istio=egressgateway -n istio-system -o jsonpath='{.items[0].metadata.name}') -c istio-proxy -n istio-system -- curl -s localhost:15000/stats | grep $MONGODB_PORT | grep ${MONGODB_HOST}.upstream_cx_total
+    cluster.outbound|<your MongoDB port>||<your MongoDB host>.upstream_cx_total: 1
+    {{< /text >}}
+
+#### Cleanup directing TLS Egress traffic through an egress gateway
+
+{{< text bash >}}
+$ kubectl delete serviceentry mongo
+$ kubectl delete gateway istio-egressgateway
+$ kubectl delete virtualservice direct-mongo-through-egress-gateway
+$ kubectl delete destinationrule egressgateway-for-mongo
+{{< /text >}}
+
+### Enable MongoDB TLS egress traffic to arbitrary wildcarded domains
+
+Sometimes you want to configure egress traffic to multiple hostnames from the same domain, for example traffic to all
+MongoDB services from `*.<your company domain>.com`. You do not want to create multiple configuration items, one per each
+and every MongoDB service in your company. To configure access to all the external services from the same domain by a
+single configuration, you use *wildcarded* hosts.
+
+In this section you configure egress traffic for a wildcarded domain, namely `*.com` and direct that traffic through
+an egress gateway. You deploy a custom egress gateway with [an additional SNI proxy](/docs/examples/advanced-gateways/wildcard-https-egress-gateway/#https-traffic-to-arbitrary-wildcarded-domains)
+which you must deploy due to current limitations of Envoy, the proxy of the standard Istio egress gateway.
+
+#### Prepare a new egress gateway with an SNI proxy
+
+In this subsection you deploy an egress gateway with an SNI proxy, in addition to the standard Istio Envoy proxy. You
+can use any SNI proxy that is capable to route traffic according to arbitrary, not-preconfigured SNI values; we used
```

Comment:
capable of routing traffic

Reply:
Done in ddc0da9cac224db35f685a67c569945dfc418870.